pandora-client-web/src/components/profileScreens/profileDescription.tsx

```diff
@@ -0,0 +1,24 @@
+import type { ReactElement } from 'react';
+import { RenderedLink } from '../../ui/screens/spaceJoin/spaceJoin';
+import React from 'react';
+
+export function ProfileDescription({ contents }: { contents: string; }): ReactElement {
+	const segments = contents.split(/(https?:\/\/\S+)/);
+	return (
+		<div>
+			{ segments.map((segment, index) => <DescriptionSegment key={ index } segment={ segment } index={ index } />) }
+		</div>
+	);
+}
+
+function DescriptionSegment({ segment, index }: { segment: string; index: number; }): ReactElement {
+	if (segment.match(/^https?:\/\//)) {
+		const url = new URL(segment);
+		return (
+			<RenderedLink key={ index } index={ index } url={ url } />
+		);
```

Comment:
I feel it would be best to wrap this in a try/catch as `new URL('https://%')` would throw and cause the component to crash

`RenderChatPart` would also need it, it was just more noticeable when you edit a profile rather then pasting a link in chat

```ts
		try {
			const url = new URL(segment);
			return (
				<RenderedLink key={ index } index={ index } url={ url } />
			);
		} catch { /**/ }
```